search-sentiments-map/public/scripts/current-data-script.js

```diff
@@ -12,37 +12,63 @@
 // License for the specific language governing permissions and limitations under
 // the License.
 
-// The current trend a user is viewing.
-let currentTrend = '';
+// JSON object for the data that is currently displayed, including topic,
+// timestamp, and custom search data by country.
+let currentSearchData = '';
+// JSON object for the trends that are currently displayed and their timestamp.
+let topTrends = '';
 
-// The custom search data for the trend that the user is viewing.
-let currentCustomSearchData = '';
+/** Returns current trend user is viewing. */
+function getTopTrends() {
+  return topTrends;
+}
 
-/** Returns the current trend that the user is viewing. */
-function getCurrentTrend() {
-  return currentTrend;
+/** Returns current custom search data for trend that user is viewing. */
+function getCurrentSearchData() {
+  return currentSearchData;
 }
 
 /** 
- * Returns current custom search data for the trend that the user is viewing.
+ * Retrieves relevant data for new trend and reconstructs the map with new
+ * data.
+ * @param {string} trend New trend to get data for.
  */
-function getCurrentCustomSearchData() {
-  return currentCustomSearchData;
+function setNewTrend(trend) {
+  fetch('/search/' + trend)
+      .then(resultsJsonArray => resultsJsonArray.json()).then(topicData => {
+        currentSearchData = topicData;
+      }).then(() => {
+        // Reload map with new sentiment or search interest data and relevant
+        // coloring.
+        loadCountryDataByMode();
+      });
 }
 
 /** 
  * Retrieves relevant data for new trend and reconstructs map with new data.
- * @param {string} trend New trend to get data for.
+ * @param {string} topic New topic to get data for.
+ * @param {Array} countries Countries to get data for.
  */
-function setNewTrend(trend) {
-  currentTrend = trend;
+function setUserSearchTopic(topic, countries) {
+  fetch('/search/' + topic + '/' + JSON.stringify(countries))
+      .then(response => response.json())
+      .then(topicResults => {
+        currentSearchData = topicResults;
+      }).then(() => {
+          // Reload map with new sentiment or search interest data and relevant
+          // coloring.
+          loadCountryDataByMode();
+          document.getElementById('submit-user-topic').innerHTML = 'Submit';
+          document.getElementById('submit-user-topic').disabled = false;
```

Comment:
What would be the correct indentation here? 2 blocks from . or from } ?

Reply:
I have no clue at all. I could just move the .then to the next line to be safe